src/main/java/org/openhab/tools/analysis/checkstyle/api/AbstractStaticCheck.java

```diff
@@ -68,14 +70,20 @@
      * @param searchedText - the text that we are looking for
      * @param startLineNumber - the line number from which the search starts exclusive, to start the
      *            search of the beginning of the text the startLineNumber should be 0
+     * @param lineFormatterCallback - a callback function to format the line before trying to find the searched text
      * @return the number of the line starting from 1, where the searched text occurred for the first
      *         time
      * @throws NoResultException when no match was found
      */
-    protected int findLineNumber(FileText fileContent, String searchedText, int startLineNumber)
-            throws NoResultException {
+    protected int findLineNumberFormatted(FileText fileContent, String searchedText, int startLineNumber,
```

Comment:
Imho this is not the best way forward it looks a little over designed to me with this callback logic, isn't there an easy way to solve this?

Reply:
It also seems that source from which we receive the string has not the same origin as lines we are searching through which seems dangerous for a lot of reasons imho.

Reply:
The sources differ is because the **searchedText** is generated by the markdown parsing logic and the **fileContent** is the raw text in the markdown file.

Reply:
Yes, I did understand that, but isn't this part of the problem and the thing we would actually like to resolve, can't we just request the parsed text from markdown and search through that?>

Reply:
@martinvw I will try to jump in here, because we discussed it in the office with @lpapazow and @doandzhi : Originally the check is reporting errors into the raw markdown file and for this reason we want to find the exact line in the raw file and report the error there. So if we get the parsed text and report the errors into its lines, we will probably hit an inconsistent line. For this reason we decided to parse a bit the special symbols (taken from the parser) in order to find the correct raw content. What do you think, are you tend to agree or disagree ?

Reply:
I understand what we try to achieve but it feels a little fragile to me, in most languages there are more character which can be escaped than the ones you have to escape so mismatches are possible. Without any proper tests that is hard to tell without me diving into the code myself. 

I found an issue referring to the actual underlying problem that the markdown `Node`s do not have a reference back to the original location: https://github.com/atlassian/commonmark-java/issues/1

Given that the issue is there, seems it cannot be done (easily) so we might go for this current solution but not without proper tests.

Reply:
Note that good example where escaping is not always performed can be seen here:

http://docs.openhab.org/addons/bindings/netatmo/readme.html
https://github.com/openhab/openhab2-addons/edit/master/addons/binding/org.openhab.binding.netatmo/README.md on line 49 there is a unescaped _ which does not cause problems in some views but in others it does. Docs renders fine, github view renders fine but diff's don't.